src/util/simplify_expr_class.h

```diff
@@ -140,8 +140,8 @@ class simplify_exprt
   bool simplify_update(exprt &expr);
   bool simplify_index(exprt &expr);
   bool simplify_member(exprt &expr);
-  bool simplify_byte_update(byte_update_exprt &expr);
-  bool simplify_byte_extract(byte_extract_exprt &expr);
+  resultt<> simplify_byte_update(const byte_update_exprt &);
+  resultt<> simplify_byte_extract(const byte_extract_exprt &);
```

Comment:
I should have raised this earlier: all the updated interfaces should use `NODISCARD` to make sure we never accidentally forget that the return value must now be consumed.

Reply:
Indeed; will make separate PR.